src/DataMigration/DataMigration.Autorest/custom/Cmdlets/New-AzDataMigrationSqlServerSchema.ps1

```diff
@@ -0,0 +1,159 @@
+
+        [Parameter(ParameterSetName='CommandLine', HelpMessage='Optional. Location of an editable TSQL schema script. Use this parameter only with DeploySchema Action.')]
+        [System.String]
+        ${InputScriptFilePath},
+
+        [Parameter(ParameterSetName='CommandLine', HelpMessage='Optional. Default: %LocalAppData%/Microsoft/SqlSchemaMigrations) Folder where logs will be written and the generated TSQL schema script by GenerateScript Action.')]
+        [System.String]
+        ${OutputFolder},
+
+        [Parameter(ParameterSetName='ConfigFile', Mandatory, HelpMessage='Path of the ConfigFile')]
+        [System.String]
+        ${ConfigFilePath},
+
+        [Parameter()]
+        [Microsoft.Azure.PowerShell.Cmdlets.DataMigration.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Returns true when the command succeeds
+        ${PassThru}
+    )
+
+    process 
+    {
+        try 
+        {
+            $OSPlatform = Get-OSName
+
+            if(-Not $OSPlatform.Contains("Windows"))
+            {
+                throw "This command cannot be run in non-windows environment"
+                Break;
+            }
+
+            #Defining Default Output Path
+            $DefaultOutputFolder = Get-DefaultSqlServerSchemaOutputFolder
+
+            #Defining Base and Exe paths
+            $BaseFolder = Join-Path -Path $DefaultOutputFolder -ChildPath Downloads;
+            $ExePath = Join-Path -Path $BaseFolder -ChildPath SchemaMigration.Console.csproj\SqlSchemaMigration.exe;
+
+            #Checking if BaseFolder Path is valid or not
+            if(-Not (Test-Path $BaseFolder))
+            {
+                $null = New-Item -Path $BaseFolder -ItemType "directory"
+            }
+
+            #Testing Whether Console App is downloaded or not
+            $TestExePath =  Test-Path -Path $ExePath;
+
+            #Downloading and extracting SchemaMigration Zip file
+            if(-Not $TestExePath)
+            {
+                $ZipSource = "https://migrationapps.blob.core.windows.net/schemamigration/SqlSchemaMigration.zip";
+                $ZipDestination = Join-Path -Path $BaseFolder -ChildPath "SqlSchemaMigration.zip";
+                Invoke-RestMethod -Uri $ZipSource -OutFile $ZipDestination;
+
+                Expand-Archive -Path $ZipDestination -DestinationPath $BaseFolder -Force;
+            }
+            # To do:
+            # else
+            # {
+                # check local exe version; 
+                # call storage account API to get latest version
+                # if version does not match, pop up to choose download Yes/No
+            # }
+
+            #Collecting data
+            if(('CommandLine') -contains $PSCmdlet.ParameterSetName)
+            {   
+                # Validate the passed Action
+                $actionList = @('migrateschema','generatescript','deployschema')
+                if(-Not $actionList.Contains($Action.ToLower()))
```

Comment:
If $Action only allows pre-defined values. You can use ValidateSet attribute in PowerShell. https://learn.microsoft.com/en-us/powershell/scripting/lang-spec/chapter-12?view=powershell-7.3